accounts/abi/bind/backends/blockchain.go

```diff
@@ -141,3 +144,197 @@ func (b *BlockchainContractCaller) getBlockAndState(num *big.Int) (*types.Block,
+	}
+	return b.txPool.AddLocal(tx)
+}
+
+func (b *BlockchainContractTransactor) ChainID(ctx context.Context) (*big.Int, error) {
+	return b.bc.Config().ChainID, nil
+}
+
+type BlockchainContractFilterer struct {
+	bc BlockChainForCaller
+	es *filters.EventSystem
+}
+
+// This nil assignment ensures at compile time that BlockchainContractCaller implements bind.ContractCaller.
+var _ bind.ContractFilterer = (*BlockchainContractFilterer)(nil)
+
+func NewBlockchainContractFilterer(bc BlockChainForCaller, es *filters.EventSystem) *BlockchainContractFilterer {
+	return &BlockchainContractFilterer{
+		bc: bc,
+		es: es,
+	}
+}
+
+func (b *BlockchainContractFilterer) FilterLogs(ctx context.Context, query klaytn.FilterQuery) ([]types.Log, error) {
+	// Convert the current block numbers into internal representations
+	if query.FromBlock == nil {
+		query.FromBlock = big.NewInt(b.bc.CurrentBlock().Number().Int64())
+	}
+	if query.ToBlock == nil {
+		query.ToBlock = big.NewInt(b.bc.CurrentBlock().Number().Int64())
+	}
+	from := query.FromBlock.Int64()
+	to := query.ToBlock.Int64()
+
+	state, err := b.bc.State()
+	if err != nil {
+		return nil, err
+	}
+	filter := filters.NewRangeFilter(&filterBackend{state.Database().TrieDB().DiskDB(), b.bc.(*blockchain.BlockChain)}, from, to, query.Addresses, query.Topics)
+
+	logs, err := filter.Logs(ctx)
+	if err != nil {
+		return nil, err
+	}
+	res := make([]types.Log, len(logs))
+	for i, log := range logs {
+		res[i] = *log
+	}
+	return res, nil
+}
+
+func (b *BlockchainContractFilterer) SubscribeFilterLogs(ctx context.Context, query klaytn.FilterQuery, ch chan<- types.Log) (klaytn.Subscription, error) {
+	// Subscribe to contract events
+	sink := make(chan []*types.Log)
+
+	if b.es == nil {
+		return nil, errors.New("events system not configured")
+	}
+	sub, err := b.es.SubscribeLogs(query, sink)
+	if err != nil {
+		return nil, err
+	}
+	// Since we're getting logs in batches, we need to flatten them into a plain stream
+	return event.NewSubscription(func(quit <-chan struct{}) error {
+		defer sub.Unsubscribe()
+		for {
+			select {
+			case logs := <-sink:
+				for _, log := range logs {
+					select {
+					case ch <- *log:
+					case err := <-sub.Err():
+						return err
+					case <-quit:
+						return nil
+					}
+				}
+			case err := <-sub.Err():
+				return err
+			case <-quit:
+				return nil
+			}
+		}
+	}), nil
+}
+
+type BlockchainContractBackend struct {
+	BlockchainContractCaller
+	BlockchainContractTransactor
+	BlockchainContractFilterer
+}
+
+// This nil assignment ensures at compile time that BlockchainContractCaller implements bind.ContractCaller.
+var _ bind.ContractBackend = (*BlockchainContractBackend)(nil)
+
+func NewBlockchainContractBackend(bc BlockChainForCaller, es *filters.EventSystem, tp *blockchain.TxPool) *BlockchainContractBackend {
+	return &BlockchainContractBackend{
+		BlockchainContractCaller{bc},
+		BlockchainContractTransactor{bc, tp},
+		BlockchainContractFilterer{bc, es},
+	}
```

Comment:
Can we have one unified class `BlockchainContractBackend` that implements all methods?
i.e.

```
type BlockchainContractBackend struct {
	bc     BlockChainForCaller
	txPool *blockchain.TxPool
	events *filters.EventSystem
}

var _ bind.ContractCaller = (*BlockchainContractBackend)(nil)
var _ bind.ContractTransactor = (*BlockchainContractBackend)(nil)
var _ bind.ContractFilterer = (*BlockchainContractBackend)(nil)
var _ bind.ContractBackend = (*BlockchainContractBackend)(nil)
```

Reply:
I think we can.

What you're suggesting is to remove `BlockchainContractCaller`, `BlockchainContractTransactor`, and `BlockchainContractFilterer` and move all their methods to `BlockchainContractBackend`, right?

Reply:
@2dvorak Yes, you got it correctly.

Reply:
Done in a5d598c, PTAL